src/main/java/core/basesyntax/strategy/PurchaseHandler.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void operateTransaction(FruitTransaction transaction) {
+        Storage storage = new Storage();
+        int oldQuantity = storage.get(transaction.getFruit());
+        storage.put(transaction.getFruit(), oldQuantity - transaction.getQuantity());
```

Comment:
please validate if the storage has enough quantity of fruits (and throw a runtime exception) before saving the result of the operation to the storage.